connections/walletconnect.js

```diff
@@ -0,0 +1,128 @@
+const EventEmitter = require("events");
+const WC = require("@walletconnect/browser");
+const WCQRCode = require("@walletconnect/qrcode-modal");
+const dev = process.env.NODE_ENV === "development";
+
+let WalletConnect;
+
+class WalletConnectConnection extends EventEmitter {
+  constructor(_WalletConnect, url, options) {
+    super();
+
+    this.qrcode =
+      typeof options.qrcode === "undefined" || options.qrcode !== false;
+
+    WalletConnect = _WalletConnect || WC;
+
+    setTimeout(() => this.create(url, options), 0);
+  }
+
+  create(url, options) {
+    try {
+      this.wc = new WalletConnect({ bridge: url });
+    } catch (e) {
+      return this.emit("error", e);
```

Comment:
Could we be more explicit in logging the "error"?

Perhaps

`return this.emit("error creating the url", e);`

Thoughts? Thanks! ☺️

Reply:
In this case 'error' is the event name, the error instance will contain the error message.

Reply:
Excellent, thanks for the context 👌